### PR TITLE
Add integration IAST and CX-flow

### PR DESCRIPTION
**Pull request by kecven**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](https://github.com/checkmarx-ltd/open-source-template/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/checkmarx-ltd/open-source-template/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

Add integration IAST and CX-flow

### References

https://github.com/checkmarx-ltd/cx-flow/pull/724

### Testing

src/test/resources/cucumber/features/integrationTests/cli/iastCli.feature

### Checklist

- [x] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentation is a Wiki Update, please indicate desired changes within PR MD Comment*
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used

### CODE REVIEW

**Comment by cxflowtestuser** [comment]
Scan submitted to Checkmarx

**Comment by cxflowtestuser** [comment]
Logo<br />**Checkmarx SAST - Scan Summary & Details**
### Cx-SAST Summary
Total of 1398 vulnerabilities<br />High&nbsp;**0 High**<br />Medium&nbsp;**2 Medium**<br />Low&nbsp;**1267 Low**<br />Info&nbsp;**129 Info**<br />
### Violation Summary
Medium&nbsp;**1** **Medium**<br />
[View more details on Checkmarx UI](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1061606&projectid=2518)
### Cx-SAST Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---
[44](https://github.com/checkmarx-ltd/cx-flow/blob/pr-iast-1849-integration_iast_and_jira/src/main/java/com/checkmarx/flow/controller/IastController.java#L44) |Medium|SSRF|src/main/java/com/checkmarx/flow/controller/IastController.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1061606&projectid=2518&pathid=1434)
***
Logo
**Checkmarx SCA - Scan Summary & Details**
### Cx-SCA Summary
**Total Packages Identified**: **266**
**Scan Risk Score**: **9.80**

High&nbsp;**9** **High** **severity vulnerabilities**
Medium&nbsp;**2** **Medium** **severity vulnerabilities**
Low&nbsp;**1** **Low** **severity vulnerabilities**
[View more details on Checkmarx UI](https://eu.sca.checkmarx.net/#/projects/3e316ee7-28b0-4dc4-ac7e-5328194b2edd/reports/069248f4-8079-4517-88ca-a4845510efa2)
### Cx-SCA vulnerability result overview
|Vulnerability ID|Package|Severity|CVSS score|Publish date|Current version|Recommended version|Link in CxSCA|Reference – NVD link|
---|---|---|---|---|---|---|---|---
`CVE-2016-1000031`|commons-fileupload:commons-fileupload|HIGH|9.8|2016-10-25T14:29:00|1.0|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/3e316ee7-28b0-4dc4-ac7e-5328194b2edd/reports/069248f4-8079-4517-88ca-a4845510efa2/vulnerabilities/CVE-2016-1000031%3AMaven-commons-fileupload%3Acommons-fileupload-1.0/vulnerabilityDetails)|[CVE-2016-1000031](https://nvd.nist.gov/vuln/detail/CVE-2016-1000031)
`CVE-2016-1000027`|org.springframework:spring-web|HIGH|9.8|2020-01-02T23:15:00|5.2.10.RELEASE|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/3e316ee7-28b0-4dc4-ac7e-5328194b2edd/reports/069248f4-8079-4517-88ca-a4845510efa2/vulnerabilities/CVE-2016-1000027%3AGradle-org.springframework%3Aspring-web-5.2.10.RELEASE/vulnerabilityDetails)|[CVE-2016-1000027](https://nvd.nist.gov/vuln/detail/CVE-2016-1000027)
`CVE-2016-1000027`|org.springframework:spring-webmvc|HIGH|9.8|2020-01-02T23:15:00|5.2.10.RELEASE|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/3e316ee7-28b0-4dc4-ac7e-5328194b2edd/reports/069248f4-8079-4517-88ca-a4845510efa2/vulnerabilities/CVE-2016-1000027%3AGradle-org.springframework%3Aspring-webmvc-5.2.10.RELEASE/vulnerabilityDetails)|[CVE-2016-1000027](https://nvd.nist.gov/vuln/detail/CVE-2016-1000027)
`CVE-2016-3092`|commons-fileupload:commons-fileupload|HIGH|7.5|2016-07-04T22:59:00|1.0|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/3e316ee7-28b0-4dc4-ac7e-5328194b2edd/reports/069248f4-8079-4517-88ca-a4845510efa2/vulnerabilities/CVE-2016-3092%3AMaven-commons-fileupload%3Acommons-fileupload-1.0/vulnerabilityDetails)|[CVE-2016-3092](https://nvd.nist.gov/vuln/detail/CVE-2016-3092)
`CVE-2014-0050`|commons-fileupload:commons-fileupload|HIGH|7.5|2014-04-01T06:27:00|1.0|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/3e316ee7-28b0-4dc4-ac7e-5328194b2edd/reports/069248f4-8079-4517-88ca-a4845510efa2/vulnerabilities/CVE-2014-0050%3AMaven-commons-fileupload%3Acommons-fileupload-1.0/vulnerabilityDetails)|[CVE-2014-0050](https://nvd.nist.gov/vuln/detail/CVE-2014-0050)
`CVE-2013-2186`|commons-fileupload:commons-fileupload|HIGH|7.5|2013-10-28T21:55:00|1.0|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/3e316ee7-28b0-4dc4-ac7e-5328194b2edd/reports/069248f4-8079-4517-88ca-a4845510efa2/vulnerabilities/CVE-2013-2186%3AMaven-commons-fileupload%3Acommons-fileupload-1.0/vulnerabilityDetails)|[CVE-2013-2186](https://nvd.nist.gov/vuln/detail/CVE-2013-2186)
`CVE-2021-25122`|org.apache.tomcat.embed:tomcat-embed-core|HIGH|7.5|2021-03-01T12:15:00|9.0.39|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/3e316ee7-28b0-4dc4-ac7e-5328194b2edd/reports/069248f4-8079-4517-88ca-a4845510efa2/vulnerabilities/CVE-2021-25122%3AGradle-org.apache.tomcat.embed%3Atomcat-embed-core-9.0.39/vulnerabilityDetails)|[CVE-2021-25122](https://nvd.nist.gov/vuln/detail/CVE-2021-25122)
`CVE-2020-17527`|org.apache.tomcat.embed:tomcat-embed-core|HIGH|7.5|2020-12-03T19:15:00|9.0.39|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/3e316ee7-28b0-4dc4-ac7e-5328194b2edd/reports/069248f4-8079-4517-88ca-a4845510efa2/vulnerabilities/CVE-2020-17527%3AGradle-org.apache.tomcat.embed%3Atomcat-embed-core-9.0.39/vulnerabilityDetails)|[CVE-2020-17527](https://nvd.nist.gov/vuln/detail/CVE-2020-17527)
`CVE-2021-25329`|org.apache.tomcat.embed:tomcat-embed-core|HIGH|7.0|2021-03-01T12:15:00|9.0.39|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/3e316ee7-28b0-4dc4-ac7e-5328194b2edd/reports/069248f4-8079-4517-88ca-a4845510efa2/vulnerabilities/CVE-2021-25329%3AGradle-org.apache.tomcat.embed%3Atomcat-embed-core-9.0.39/vulnerabilityDetails)|[CVE-2021-25329](https://nvd.nist.gov/vuln/detail/CVE-2021-25329)
`CVE-2021-24122`|org.apache.tomcat.embed:tomcat-embed-core|MEDIUM|5.9|2021-01-14T15:15:00|9.0.39|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/3e316ee7-28b0-4dc4-ac7e-5328194b2edd/reports/069248f4-8079-4517-88ca-a4845510efa2/vulnerabilities/CVE-2021-24122%3AGradle-org.apache.tomcat.embed%3Atomcat-embed-core-9.0.39/vulnerabilityDetails)|[CVE-2021-24122](https://nvd.nist.gov/vuln/detail/CVE-2021-24122)
`CVE-2020-17521`|org.codehaus.groovy:groovy|MEDIUM|5.5|2020-12-07T20:15:00|2.5.13|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/3e316ee7-28b0-4dc4-ac7e-5328194b2edd/reports/069248f4-8079-4517-88ca-a4845510efa2/vulnerabilities/CVE-2020-17521%3AGradle-org.codehaus.groovy%3Agroovy-2.5.13/vulnerabilityDetails)|[CVE-2020-17521](https://nvd.nist.gov/vuln/detail/CVE-2020-17521)
`CVE-2013-0248`|commons-fileupload:commons-fileupload|LOW|3.3|2013-03-15T20:55:00|1.0|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/3e316ee7-28b0-4dc4-ac7e-5328194b2edd/reports/069248f4-8079-4517-88ca-a4845510efa2/vulnerabilities/CVE-2013-0248%3AMaven-commons-fileupload%3Acommons-fileupload-1.0/vulnerabilityDetails)|[CVE-2013-0248](https://nvd.nist.gov/vuln/detail/CVE-2013-0248)

File: src/main/java/com/checkmarx/flow/CxFlowRunner.java [hunk 1]
```diff
@@ -188,8 +197,13 @@ private void commandLineRunner(ApplicationArguments args) throws ExitThrowable {
         CxPropertiesBase cxProperties = cxScannerService.getProperties();
 
         if (((ScanUtils.empty(namespace) && ScanUtils.empty(repoName) && ScanUtils.empty(branch)) &&
-                ScanUtils.empty(application)) && !args.containsOption(BATCH_OPTION)) {
-            log.error("Namespace/Repo/Branch or Application (app) must be provided");
+                ScanUtils.empty(application)) && !args.containsOption(BATCH_OPTION) && !args.containsOption(IAST_OPTION)) {
```
Comment (kecven) [inline src/main/java/com/checkmarx/flow/CxFlowRunner.java:200]: we can delete IAST_OPTION from here.

File: src/main/java/com/checkmarx/flow/CxFlowRunner.java [hunk 2]
```diff
@@ -188,8 +197,13 @@ private void commandLineRunner(ApplicationArguments args) throws ExitThrowable {
...
+            log.error("Namespace/Repo/Branch/Iast or Application (app) must be provided");
+            exit(1);
+        }
+
+        if (args.containsOption(IAST_OPTION) && (scanTag == null)) {
```
Comment (kecven) [inline src/main/java/com/checkmarx/flow/CxFlowRunner.java:205]: (scanTag == null) better change to StringUtils.empty()

**Pull request merged**